src/value.c

```diff
@@ -571,36 +574,7 @@ int RSValue_Equal(const RSValue *v1, const RSValue *v2, QueryError *qerr) {
   v2 = RSValue_Dereference(v2);
 
   if (v1->t == v2->t) {
-    switch (v1->t) {
-      case RSValue_Number:
-        return cmp_numbers(v1, v2) == 0;
-        
-      case RSValue_String:
-        return cmp_strings(v1->strval.str, v2->strval.str, v1->strval.len, v2->strval.len) == 0;
-        
-      case RSValue_RedisString:
-      case RSValue_OwnRstring: {
-        size_t l1, l2;
-        const char *s1 = RedisModule_StringPtrLen(v1->rstrval, &l1);
-        const char *s2 = RedisModule_StringPtrLen(v2->rstrval, &l2);
-        return cmp_strings(s1, s2, l1, l2) == 0;
-      }
-      
-      case RSValue_Null:
-        return 1;
-
-      case RSValue_Array: {
```

Comment:
In case of equality comparison, we can compare Nulls. Maybe it isn't compatible with our semantics, though.

Reply:
It doesn't seem to break the tests. I don't think we have any specific semantics regarding NULLs, though I'd say our use of NULL is limited because it's also used as the "default" value for empty fields... we don't really make use of a discreet NULL value. There is `Undef`, but I haven't actually implemented the distinction well enough in practice